lib/dialects/postgres/query-generator.js

```diff
@@ -758,7 +753,10 @@ var QueryGenerator = {
       values = dataType.toString().match(/^ENUM\(.+\)/)[0];
     }
 
-    var sql = 'CREATE TYPE ' + enumName + ' AS ' + values + ';';
+    var schema = tableName.schema || options.schema || 'public';
+    var delimiter = tableName.delimiter || options.delimiter || '.';
+
+    var sql = 'CREATE TYPE ' + this.quoteIdentifier(schema) + delimiter + enumName + ' AS ' + values + ';';
```

Comment:
Then the change should be the other way I think - so that enumName is generated manually here. `pgEnumName` is used name places, otherwise we'd have to duplicate the logic for adding schema in front of enum name to all those functiond